File: e2e/java/src/test/java/dev/kreuzberg/e2e/MarkdownTest.java
package dev.kreuzberg.e2e;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/** E2e tests for category: markdown. */
class MarkdownTest {
    @Test
    void testMarkdownBasicConversion() {
        // HTML is always converted to markdown alongside raw HTML
        var result = Kreuzcrawl.scrape();
        assertEquals(200, result.status_code());
        assertEquals("Test", result.metadata().title());
        assertFalse(result.html().isEmpty(), "expected non-empty value");
        assertFalse(result.markdown().isEmpty(), "expected non-empty value");
        assertTrue(result.markdown().contains("Hello World"), "expected to contain: " + "Hello World");
    }

    @Test
    void testMarkdownCrawlAllPages() {
        // All crawled pages have markdown field populated
        var result = Kreuzcrawl.scrape();
        assertEquals(2, result.crawl().pages_crawled());
    }

    @Test
    void testMarkdownFitContent() {
        // Fit markdown removes navigation and boilerplate content
        var result = Kreuzcrawl.scrape();
        assertEquals(200, result.status_code());
        assertFalse(result.markdown().isEmpty(), "expected non-empty value");
    }

    @Test
    void testMarkdownHeadingsAndParagraphs() {
        // Markdown conversion preserves heading hierarchy and paragraph text
        var result = Kreuzcrawl.scrape();
        assertFalse(result.markdown().isEmpty(), "expected non-empty value");
        assertTrue(result.markdown().contains("Main Title"), "expected to contain: " + "Main Title");
    }

    @Test
    void testMarkdownLinksConverted() {
        // HTML links are converted to markdown link syntax
        var result = Kreuzcrawl.scrape();
        assertEquals(200, result.status_code());
        assertFalse(result.html().isEmpty(), "expected non-empty value");
        assertFalse(result.markdown().isEmpty(), "expected non-empty value");
        assertTrue(result.markdown().contains("Example"), "expected to contain: " + "Example");
    }

    @Test
    void testMarkdownWithCitations() {
        // Markdown includes citation conversion with numbered references
        var result = Kreuzcrawl.scrape();
        assertEquals(200, result.status_code());
        assertFalse(result.markdown().isEmpty(), "expected non-empty value");
    }

}

File: e2e/node/tests/markdown.test.ts
import { describe, it, expect } from 'vitest';
import { scrape, createEngine } from '@kreuzberg/kreuzcrawl';

describe('markdown', () => {
  it('markdown_basic_conversion: HTML is always converted to markdown alongside raw HTML', async () => {
    const engine = createEngine(null);
    const result = await scrape(engine, "");
    expect(result.statusCode).toBe(200);
    expect(result.metadata.title).toBe("Test");
    expect(result.html.length).toBeGreaterThan(0);
    expect(result.markdown.length).toBeGreaterThan(0);
    expect(result.markdown).toContain("Hello World");
  });

  it('markdown_crawl_all_pages: All crawled pages have markdown field populated', async () => {
    const engine = createEngine(null);
    const result = await scrape(engine, "");
    // skipped: field 'crawl.pages_crawled' not available on result type
  });

  it('markdown_fit_content: Fit markdown removes navigation and boilerplate content', async () => {
    const engine = createEngine(null);
    const result = await scrape(engine, "");
    expect(result.statusCode).toBe(200);
    expect(result.markdown.length).toBeGreaterThan(0);
  });

  it('markdown_headings_and_paragraphs: Markdown conversion preserves heading hierarchy and paragraph text', async () => {
    const engine = createEngine(null);
    const result = await scrape(engine, "");
    expect(result.markdown.length).toBeGreaterThan(0);
    expect(result.markdown).toContain("Main Title");
  });

  it('markdown_links_converted: HTML links are converted to markdown link syntax', async () => {
    const engine = createEngine(null);
    const result = await scrape(engine, "");
    expect(result.statusCode).toBe(200);
    expect(result.html.length).toBeGreaterThan(0);
    expect(result.markdown.length).toBeGreaterThan(0);
    expect(result.markdown).toContain("Example");
  });

  it('markdown_with_citations: Markdown includes citation conversion with numbered references', async () => {
    const engine = createEngine(null);
    const result = await scrape(engine, "");
    expect(result.statusCode).toBe(200);
    expect(result.markdown.length).toBeGreaterThan(0);
  });
});
